### ISSUE TITLE
KA10: Interrupt problem

### ISSUE THREAD

**Issue by larsbrinkhoff** [rcornwell/sims#60]
Running FED using the 340 display will cause ITS to crash.

During the course of running FED, where will be an interrupt from the 340.  PIH is set to 10 (level 4).

When the interrupt service routine returns through LPBKRT, PIH has changed to 12.  When LPBKRT+1 executes a JEN, PIH is set to 2.  But check_irq_level will not trigger a new interrupt request on level 1 because PIH is already 2, and it remains at 2 when ITS returns to user mode.  The next time user mode calls a UUO, ITS notices PIH is 2, and throws the message "UUO WITH PI IN PROGRESS".

**Comment by larsbrinkhoff** [comment]
Steps to repeat:

- Run ITS with 340 support: 340P==1 and %TT340 set on TTY 0 (console).
- Install some font.  http://lars.nocrew.org/its/fonts.dump
- Run FED.
- It prompts INPUT:
- Answer the the file name of some font.  The default directory is FONTS, and KST is default for FN2.

After having read the font file, ITS crashes with this message:

```
UUO WITH PI IN PROGRESS  
PI LEVEL 6 BUGHALT.  FIND A WIZARD OR CONSIDER TAKING A CRASH DUMP.
THE SYSTEM HAS CRASHED AND CANNOT BE REVIVED WITHOUT EXPERT ATTENTION.
IF YOU CAN'T FIND HELP, RELOAD THE SYSTEM.
YOU ARE NOW IN DDT.
BUGPC/   CAI UUOH0+2   $Q-2/   CONSZ PI,TTYST2+4

**Comment by larsbrinkhoff** [comment]
I have now verified this against the current sims master branch as of now.  Commit id bbe8a6252bb6a5ce666664a8e535c48ea08721aa.

This is a clean build of the simulator, and the ITS repository.  With the addition of font files.

**Comment by larsbrinkhoff** [comment]
Working around the CONO in the second interrupt instruction fixes this.  But there's another error instead.

Can I close this and open a new issue for the other error?

**Comment by rcornwell** [comment]
If you want me to fix the simulator to support CONO in IRQ slot keep this open, otherwise close it.

**Comment by larsbrinkhoff** [comment]
CONO in IRQ slot is ".DSTART crash" #64.  So in a way these two duplicate each other.

**Comment by rcornwell** [comment]
Looks like they might be similar. If your work around fixes #64 then they are a duplicate.

**Comment by larsbrinkhoff** [comment]
#64 has much better information about the CONO things, so I'll close this.
